pilot/pkg/model/gateway.go

```diff
@@ -253,6 +265,23 @@ func MergeGateways(gateways []gatewayWithInstances, proxy *Proxy) *MergedGateway
 						} else {
 							mergedServers[serverPort].Servers = append(mergedServers[serverPort].Servers, s)
 						}
+
+						// We have TLS settings defined and we have already taken care of unique route names
+						// if it is HTTPS. So we can construct a QUIC server on the same port. It is okay as
+						// QUIC listens on UDP port, not TCP
+						if features.EnableQUICListeners && gateway.IsEligibleForHTTP3Upgrade(s) &&
+							udpSupportedPort(s.GetPort().GetNumber(), gwAndInstance.instances) {
+							log.Debugf("Server at port %d eligible for HTTP3 upgrade. Add UDP listener for QUIC", serverPort.Number)
+							if mergedQUICServers[serverPort] == nil {
+								mergedQUICServers[serverPort] = &MergedServers{
+									Servers:   []*networking.Server{s},
+									RouteName: routeName,
```

Comment:
what if routeName changes between serverPort? Is it possible?

Reply:
Something I need to go through once again.

Reply:
It is possible because different HTTPS servers would have different route names. I went through the code again and found that `RouteName` is unnecessary for HTTPS and HTTP/3 cases. This is because we take it from `TLSServerInfo`. The field is used only for plaintext HTTP. Hence I removed this in both places.